src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/provisioning/DeviceProvisioningFragment.kt

```diff
@@ -158,6 +158,13 @@ class DeviceProvisioningFragment : Fragment() {
     override fun onPairingComplete(code: Int) {
       Log.d(TAG, "onPairingComplete: $code")
 
+      // In IP commissioning, commissioning complete will already be called at this point, and the
+      // next fragment will be shown. As a result, this fragment will be in a destroyed state and
+      // should not be operated on.
+      if (deviceInfo.ipAddress != null) {
```

Comment:
Should we display an error message on failure?

Reply:
showMessage() won't work here as the fragment wouldn't be attached to the activity. We will still have the log for the error code though.